rasa/shared/core/training_data/story_writer/yaml_story_writer.py

```diff
@@ -228,7 +229,30 @@ def process_user_utterance(
             entities = []
             for entity in user_utterance.entities:
                 if "value" in entity:
-                    entities.append(OrderedDict([(entity["entity"], entity["value"])]))
+                    if hasattr(user_utterance, "inline_comment_for_entity"):
+                        for predicted in user_utterance.predicted_entities:
+                            if predicted["start"] == entity["start"]:
+                                if predicted["entity"] != entity["entity"]:
```

Comment:
I added this check because otherwise if I have more than one entities and one is correctly predicted and the other is wrongly predicted then it does not print the correct one 🤦🏽‍♀️ 

This happens because with the check `if hasattr(user_utterance, "inline_comment_for_entity"):` I get always `True` if : 
- I have at least one wrongly predicted entity but I don't know which one it is beforehand. 
- I have all entities correct and the `intent` is wrong because this just checks if the method is there and it is there if the object is of the class `WronglyClassifiedUserUtterance`.

These two cases give me also `True` for the `if hasattr(user_utterance, "inline_comment"):` check and that is why I added also an extra check on intent on line 209.

Reply:
I think this logic should not be here. The `YAMLStoryWriter` should not have to know how to determine in an entity prediction is correct.
What if we changed `inline_comment_for_entity` to return `None` if there is no comment (i.e. it was correct), and then we just only add it if it returns something?

Reply:
True! I did not think about which module should know what. Good point! This is what they call [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) ?